### PR TITLE
chore: Break up Icons into Chunks, Fix IE 11 event issue

### PR DESCRIPTION
**Pull request by zero0Halo**
feature(icons): Added BulletList, Calendar, Camera, Cart, Cashback

feature(icons): Added Chat, Check, CheckSmall, ChevronDown, ChevronDownSmall

feature(icons): Added Chat, Chevron (left, right, up), Chevron Small (left, right, up), Clock, Close, CloseSmall, CommentMore

feature(icons): Added CreditCard, CrossHairs, Cut, Disabled, Dislike

feature(icons): Added Download, Ellipses, EllipsesVertical, Envelope, EnvelopeOpen

feature(icons): Added Error, Expand, Gear, GiftCard, Grid

I also added formatting to the All Icons story so the icons are in a grid pattern

feature(icons): Added Hamburger, Heart, HeartOutline, Home, Info

feature(icons): Added InfoOutline, Laptop, Lighting, Like

feature(icons): Added List, Lock, Map, Marker, MarkerOutline

feature(icons): Added Mobile, News, Pencil, Play, Plus

feature(icons): Added PlusSmall, Print, Question, QuestionOutline, Refresh

feature(icons): Updated Error, added RetailMeNotLogo, SadFace, Search, Share, Sliders

feature(icons): Added Star, StarHalf, StarOutline, Success, SuccessOutline, Tag, TagAdd, Upload

feature(icons): Testing & Niceties

* Updated the README with some basic info.
* Added tests
* Added snapshots
* Provided access to the icons as an Icon object as well as individually
* Made the All Icons story prettier

chore(Icons): Updated test to cover all of the new icon components

chore(Icons): Replaced old Icon component with new one

Updated other components that were using the original Icon component to use the new one

**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [ ] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.

---------
**Outline your feature or bug-fix below**

This refactors the Icon component to hopefully allow for better tree-shaking, i.e. Using one icon doesn't import all of them into a bundle.

### CODE REVIEW

File: src/Icon/Icon.component.spec.tsx [hunk 1]
```diff
@@ -1,25 +1,15 @@
...
 import * as renderer from 'react-test-renderer';
 // COMPONENT
-import Icon from './Icon.component';
-// ENZYME
-import { mount } from 'enzyme';
-
-// TEST SETUP
-const subject = <Icon type="hamburger" />;
+import Icon from './';
+// import { Scale } from './utils';
```
Comment (bjohnsonrmn) [inline src/Icon/Icon.component.spec.tsx]: Did you mean to leave this comment in?

File: src/Icon/utils.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components';
+
+type ScaleFactors = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';
+
+export interface IconSVGProps {
+    scale?: ScaleFactors;
```
Comment (bjohnsonrmn) [inline src/Icon/utils.ts:6]: Perhaps `scale` and `color` should be required. Can we render an icon without them?

File: src/Icon/README.md [hunk 3]
```diff
@@ -1 +1,14 @@
 # Icon
+
+The Icon component is actually an object which holds all Icon components. 
+Trying to render `<Icon />` will result in an error, but for example, `<Icon.ArrowBack />` would not.
+
+## Usage
```
Comment (bjohnsonrmn) [inline src/Icon/README.md:6]: Does this usage example show how to use only 1 icon so that the JS for all the others is not included?

File: src/Icon/Icons/ArrowBack.component.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// REACT
+import * as React from 'react';
+import classNames from 'classnames';
+import { IconSVGProps, Scale, StyledIcon } from '../utils';
```
Comment (bjohnsonrmn) [inline src/Icon/Icons/ArrowBack.component.tsx]: This seems to be 1 of 2 components (`ArrowForward` is the other) that doesn't use `DefaultColor` and `DefaultScale`. Is that intentional?

**Review by bjohnsonrmn — COMMENTED** [review]
@zero0Halo I had a few small questions, but overall this looks good to me 👍

File: src/Icon/utils.ts (hunk 2 above)
Comment (sharkparty) [reply src/Icon/utils.ts:6]: This did make testing unnecessarily difficult. We can default scale to 16px and color to #222

File: src/Icon/README.md (hunk 3 above)
Comment (sharkparty) [reply src/Icon/README.md:6]: As they're all the same, I'm kosher with this.

File: src/Icon/Icons/ArrowBack.component.tsx (hunk 4 above)
Comment (sharkparty) [reply src/Icon/Icons/ArrowBack.component.tsx]: +1

File: src/AutoComplete/AutoComplete.stories.tsx [hunk 5]
```diff
@@ -25,16 +25,13 @@ storiesOf('Components/AutoComplete', module)
...
-                dataSource={[1, 2, 3, 4]}
-            />
+            <AutoComplete prefix={<Icon.Search />} dataSource={[1, 2, 3, 4]} />
             <p>AutoComplete 2</p>
             <AutoComplete
                 size="large"
                 placeholder="Stores, brands, categories, etc."
                 dataSource={[1, 2, 3, 4]}
-                prefix={<Icon type="search" />}
+                prefix={<Icon.Search />}
```
Comment (sharkparty) [inline src/AutoComplete/AutoComplete.stories.tsx:34]: Thanks for updating all of the stories that use this component!

File: src/Icon/Icon.component.spec.tsx (hunk 1 above)
Comment (sharkparty) [reply src/Icon/Icon.component.spec.tsx]: +1

File: src/Icon/utils.ts (hunk 2 above)
Comment (zero0Halo) [reply src/Icon/utils.ts:6]: Yes we can render icons w/o the user passing in anything, it just defaults to 'md' and '#222' as Kam noted. I think you did find two components where I failed to put in the defaults though, so I'll fix that. I will update the readme to note what the defaults are.

File: src/Icon/README.md (hunk 3 above)
Comment (zero0Halo) [reply src/Icon/README.md:6]: I was considering providing a second example, but for right now I'm only exporting `Icon` to be used. Since there is some uncertainty as to whether using `<Icon.Whatever /> ` will tree shake, I will go back and update it so that all Icons are exported individually. It will make our main index.ts file gross, but best to hedge our bets.

File: src/Icon/Icons/ArrowBack.component.tsx (hunk 4 above)
Comment (zero0Halo) [reply src/Icon/Icons/ArrowBack.component.tsx]: Good catch. I'll fix that up.

**Comment by bjohnsonrmn** [comment]
@zero0Halo once you update exports so that someone can import or use only a few icons, and only those icons get included in the JS and make sure all the icons have the defaults set, this looks good to merge to me ✅

File: src/Icon/README.md [hunk 6]
```diff
@@ -1 +1,31 @@
...
+    import { Icon } from '@retailmenot/anchor';
+    
+    const example = props => (
+        <Icon.ArrowBack />
+    );
+    
+Alternatively, each icon can be imported individually.
```
Comment (bjohnsonrmn) [inline src/Icon/README.md:13]: Thanks for adding this ❤️

File: src/Icon/README.md [hunk 7]
```diff
@@ -1 +1,31 @@
...
+    
+    import { IconArrowBack } from '@retailmenot/anchor';
+    
+    const example = props => (
+        <IconArrowBack />
+    );    
+    
+## Settings/Props
+
+The component can accept 3 `props`:
```
Comment (bjohnsonrmn) [inline src/Icon/README.md:23]: This is also helpful 👍

**Review by bjohnsonrmn — APPROVED** [review]
@zero0Halo looks good to me ✅
